torch/functional.py

```diff
@@ -15,6 +15,14 @@
 Tensor = torch.Tensor
 from torch import _VF
 
+# Set a global declaring that we have opt_einsum
+from importlib.util import find_spec as _find_spec
+if _find_spec('opt_einsum') is not None:
+    import opt_einsum as _opt_einsum  # type: ignore[import]
+else:
+    _opt_einsum = None
```

Comment:
Imo `try:except:` is more pythonic way than `if :else`
```suggestion
try:
    import opt_einsum as _opt_einsum  # type: ignore[import]
except ModuleNotFoundError:
    _opt_einsum = None
```